agent/consul/leader_connect_ca.go

```diff
@@ -537,10 +549,11 @@ func (c *CAManager) primaryInitialize(provider ca.Provider, conf *structs.CAConf
 	if err != nil {
 		return err
 	}
-	if activeRoot != nil && needsSigningKeyUpdate {
-		c.logger.Info("Correcting stored SigningKeyID value", "previous", rootCA.SigningKeyID, "updated", expectedSigningKeyID)
+	if activeRoot != nil && rootUpdateRequired {
+		c.logger.Info("Correcting stored CARoot values",
+			"previous-signing-key", rootCA.SigningKeyID, "updated-signing-key", expectedSigningKeyID)
```

Comment:
might've gotten confused but the control flows here but wouldn't `rootCA.SigningKeyID` & `expectedSigningKeyID` be the same by the time we reach lines 553-54?

Since on line 531 we set:

```
rootCA.SigningKeyID = expectedSigningKeyID
```

did I miss something obvious 🤔 ?

Reply:
Ya, I found this `if / else if` to be not super obvious.

All of this logic (from about line 523) is about fixing existing data stored in raft and the state store. So line 531 is where we fix the local reference, but we still need to persist that fix back into raft with `ApplyCARequest`.

So here on line 553 it is correct in the local variable, but we are logging that we are about to (later on below the `if / else if`) persist it.

I would like to refactor this some more. #11687 makes some more changes in this area (for a different reason), and it moves the `activeRoot` call earlier in the function. I think once that is done we can probably refactor this to make it a bit more obvious that the logic is:

```
if nothingToFix {
   return
}

applyFixesToRaft
return
```

Reply:
I think that makes sense and that I'm almost there!

So won't the logging actually log two identical values?

Reply:
Ah, I see what you are saying now. Yes! I did mess up the logging. I've moved the log up to the if block where we check if it needs update, so it should log the correct thing now.

That allowed me to remove the `else if` and make it just an `if`, which is also nice.